docs/architecture/adr-038-state-listening.md

```diff
@@ -269,32 +289,14 @@ func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeg
 
 	...
 
-	// call the hooks with the BeginBlock messages
-	wg := new(sync.WaitGroup)
-	for _, streamingListener := range app.abciListeners {
-		streamingListener := streamingListener // https://go.dev/doc/faq#closures_and_goroutines
-		if streamingListener.HaltAppOnDeliveryError() {
-			// increment the wait group counter
-			wg.Add(1)
-			go func() {
-				// decrement the counter when the go routine completes
-				defer wg.Done()
-				if err := streamingListener.ListenBeginBlock(app.deliverState.ctx, req, res); err != nil {
-					app.logger.Error("BeginBlock listening hook failed", "height", req.Header.Height, "err", err)
-					app.halt()
-				}
-			}()
-		} else {
-			// fire and forget semantics
-			go func() {
-				if err := streamingListener.ListenBeginBlock(app.deliverState.ctx, req, res); err != nil {
-					app.logger.Error("BeginBlock listening hook failed", "height", req.Header.Height, "err", err)
-				}
-			}()
-		}
-	}
-	// wait for all the listener calls to finish
-	wg.Wait()
+  defer func() {
+    // call the hooks with the BeginBlock messages
+    for _, streamingListener := range app.abciListeners {
```

Comment:
what is the reasoning to do this synchronous, apposed to concurrent before?

Reply:
ah nvm, figured out down lower.

Reply:
basically, it's more natural to propagate errors by default, the streamer implementation can chose to don't return errors, like set the `stop-node-on-error` to `false` for file streamer, it won't return any error, just output an error log.